### PR TITLE
Quickfixes should be available at the line level

### LINKED ISSUE #2130: Quickfixes should be available at the line level
Given the following code:
```java
public class Foo {
    public static void main(String[] args) {
        System.out.println(greeting());
    }
}
```
`greeting` is underlined in red, because the method doesn't exist. If you put your cursor on it, and list the quickfixes (`ctrl+.`), you get to create the missing method.
Now it would be nice to click anywhere on that line to also see the same quickfix, like Intellij or Eclipse.

cc. @maxandersen

### PR DESCRIPTION
**Pull request by snjeza**
Fixes #2130 
Requires https://github.com/eclipse/eclipse.jdt.ls/pull/1908

Signed-off-by: Snjezana Peco <snjezana.peco@redhat.com>

### CODE REVIEW

**Review by rgrunber — APPROVED** [review]
This looks fine to me. Once we merge the LS side I think we can merge this also.

File: README.md [hunk 1]
```diff
@@ -194,6 +194,10 @@ The following settings are supported:
 * `java.settings.url` : Specifies the url or file path to the workspace Java settings. See [Setting Global Preferences](https://github.com/redhat-developer/vscode-java/wiki/Settings-Global-Preferences)
 * `java.symbols.includeSourceMethodDeclarations` : Include method declarations from source files in symbol search. Defaults to `false`.
 
+New in 0.83.0:
```
Comment (testforstephen) [inline README.md:197]: version info is outdated.

Comment (snjeza) [reply README.md]: Fixed.

File: package.json [hunk 2]
```diff
@@ -817,6 +817,12 @@
           "markdownDescription": "Include method declarations from source files in symbol search.",
           "default": false,
           "scope": "window"
+        },
+        "java.quickfix.showAll": {
+          "type": "boolean",
+          "default": false,
```
Comment (testforstephen) [inline package.json:823]: this feature is useful and can improve the feature discovery of quick fixes. Can we make the default value to be "**true**"?

**Comment by fbricon** [comment]
`java.quickfix.showAll` is misleading IMHO. how about: 

java.quickfix.showAt=problem|line

File: package.json (hunk 2 above)
Comment (snjeza) [reply package.json]: Fixed.

**Comment by snjeza** [comment]
> java.quickfix.showAll is misleading IMHO. how about:
java.quickfix.showAt=problem|line

Fixed.

**Comment by snjeza** [comment]
@fbricon @rgrunber @testforstephen I have updated the PR.

File: package.json [hunk 3]
```diff
@@ -817,6 +817,16 @@
...
           "scope": "window"
+        },
+        "java.quickfix.showAt": {
+          "type": "string",
+          "enum": [
+            "line",
+            "problem"
+          ],
+          "default": "line",
+          "description": "Show All Quick Fixes at the line.",
```
Comment (fbricon) [inline package.json:828]: Show quickfixes at the problem or line level

Comment (snjeza) [reply package.json]: Fixed.

**Pull request merged**